File: daily/257-Binary-tree-paths.py
class Solution:
    def binaryTreePaths(self, root: Optional[TreeNode]) -> List[str]:
        results = []
        def dfs(node, path):
            if not node:
                return
            
            path.append(node.val)
            if not node.left and not node.right:
                leafPath = "->".join(str(val) for val in path)
                results.append(leafPath)
            dfs(node.left, path)
            dfs(node.right, path)

            path.pop()
        dfs(root, [])
        return results 

File: daily/Binary Tree/257-Binary-tree-paths.go
package daily

import (
	"strconv"
	"strings"
)

func binaryTreePaths(root *TreeNode) []string {
	results := []string{}
	path := []int{}

	var dfs func(node *TreeNode)
	dfs = func(node *TreeNode) {
		if node == nil {
			return
		}

		path = append(path, node.Val)
		if node.Left == nil && node.Right == nil {
			parts := make([]string, len(path))
			for i, v := range path {
				parts[i] = strconv.Itoa(v)
			}
			results = append(results, strings.Join(parts, "->"))
		}

		dfs(node.Left)
		dfs(node.Right)

		path = path[:len(path)-1]
	}

	dfs(root)
	return results
}
